### PR TITLE
Migrate to Modern Dependencies

### PR DESCRIPTION
**Pull request by X-sam**
This PR brings the following updates:
- Migrate from Qt5 to Qt6
- Update supported Python versions to 3.8+ [instead of locked to 3.8]
- Upgrade PyInstaller from 4.4 to 6.4, the latest at the time of this PR.
- Address some inconsistencies in app naming requirements on mac

This, along with a sister PR on AYAB-Desktop (AllYarnsAreBeautiful/ayab-desktop#572), should fix release builds for us going forward.

### CODE REVIEW

File: requirements.txt
Comment (t0mpr1c3) [inline requirements.txt:2]: PyQt5==5.15.9 in https://github.com/AllYarnsAreBeautiful/ayab-desktop/pull/572 ?

File: requirements.txt
Comment (t0mpr1c3) [inline requirements.txt:1]: PyQt5==5.15.9 in https://github.com/AllYarnsAreBeautiful/ayab-desktop/pull/572 ?

Comment (t0mpr1c3) [inline requirements.txt]: PyQt5==5.15.9 in https://github.com/AllYarnsAreBeautiful/ayab-desktop/pull/572 ?

Comment (t0mpr1c3) [reply requirements.txt]: Is there any way to get away from the PyQt5 dependency? It would be nice not to have dependencies on both PyQt5 and PySide6.

**Comment by dl1com** [comment]
lgtm, let's wait for @t0mpr1c3 's approval before merging.

**Review by t0mpr1c3 — APPROVED** [review]

**Review by dl1com — APPROVED** [review]

**Pull request merged**
